pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -87,20 +97,79 @@ func (c *APIClient) connect() error {
 	return err
 }
 
-func parseKubeConfig(fpath string) (*k8s.Config, error) {
-	yamlFile, err := ioutil.ReadFile(fpath)
+// ParseKubeConfig reads and unmarcshals a kubeconfig file
+// in an object ready to use. Exported for integration testing.
+func ParseKubeConfig(fpath string) (*k8s.Config, error) {
+	// TODO: support yaml too
+	jsonFile, err := ioutil.ReadFile(fpath)
 	if err != nil {
 		return nil, err
 	}
 
 	config := &k8s.Config{}
-	err = yaml.Unmarshal(yamlFile, config)
+	err = json.Unmarshal(jsonFile, config)
 	return config, err
 }
 
+// ComponentStatuses returns the component status list from the APIServer
 func (c *APIClient) ComponentStatuses() (*v1.ComponentStatusList, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	return c.client.CoreV1().ListComponentStatuses(ctx)
+}
+
+// ConfigMapTokenFetcher returns the value of the `tokenValue` from the `tokenKey` in the ConfigMap `configmaptokendca` if its timestamp is less than tokenTimeout old.
+func (c *APIClient) ConfigMapTokenFetcher(tokenKey string) (string, bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+	tokenConfigMap, err := c.client.CoreV1().GetConfigMap(ctx, configMapDCAToken, defaultNamespace)
+	if err != nil {
+		return "", false, err
```

Comment:
`false` is pretty non-descript, I would've gone with a standard error clients can check and ignore, like https://github.com/DataDog/datadog-agent/blob/84538338fd2e8210e20b46a2d8747afea571b4c7/pkg/tagger/collectors/types.go#L25

Reply:
you could also create an error type for when the configmap is here but too old